specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/common.json

```diff
@@ -69,6 +69,59 @@
       },
       "description": "The key vault server error.",
       "x-nullable": true
+    },
+    "LifetimeActions": {
+      "properties": {
+        "trigger": {
+          "$ref": "#/definitions/LifetimeActionsTrigger",
+          "description": "The condition that will execute the action."
+        },
+        "action": {
+          "$ref": "#/definitions/LifetimeActionsType",
+          "description": "The action that will be executed."
+        }
+      },
+      "description": "Action and its trigger that will be performed by Key Vault over the lifetime of a key or secret."
+    },
+    "LifetimeActionsTrigger": {
+      "properties": {
+        "timeAfterCreate": {
+          "type": "string",
+          "description": "Time after creation to attempt to rotate. It only applies to rotate. It will be in ISO 8601 duration format. Example: 90 days : \"P90D\" "
+        },
+        "timeBeforeExpiry": {
+          "type": "string",
+          "description": "Time before expiry to attempt to rotate or notify. It will be in ISO 8601 duration format. Example: 90 days : \"P90D\""
+        }
```

Comment:
Just as an FYI, we'll not currently be supporting ISO 8601 durations as anything other than a string in the Azure SDKs. We've discussed it at length and there are problems we can't easily resolve in most of our languages. We may do something in the future but not ready yet.

That's not to say don't use ISO 8601 durations - I prefer them to having to store a numerical value and unit type - but they'll just be strings in the SDK, so users are on the hook for formatting them. As such, you might consider adding a link to the IETF RFC for help. This text will, by default, be the doc comments for the SDK.

/cc @herveyw-msft as an FYI, though I think he was part of this conversation already.

Reply:
We had discussion with review board and that was approved for keys and now we extend it to other objects in vault

Reply:
Right - I was part of those talks. This wasn't feedback so much as an FYI ("Just as an FYI..."). Since I thought we punted key rotation to 7.4-preview.1, I thought there might still be time to change this if you wanted to; though, you don't have to; like I said above, I prefer ISO 8601 durations myself - no reason to reinvent durations.